sympy/matrices/expressions/tests/test_matmul.py

```diff
@@ -119,3 +119,9 @@ def test_collapse_MatrixBase():
 
 def test_refine():
     assert refine(C*C.T*D, Q.orthogonal(C)).doit() == D
+
+def test_issue_9503():
+    from sympy import I
+    a = symbols('a')
+    M = Matrix([[1, 2 + I], [3, 4]])
+    assert transpose(MatMul(a, M)) == MatMul(transpose(a), transpose(M))
```

Comment:
As far you said `transpose(a)` should mean the same thing. In any case it should not
because for `a` we can later on substitute for values. And here `a` should not be referred to as being a scalar, i guess.
like.
```
>>> from sympy.matrices.expressions.transpose import transpose as mat_trans
>>> mat_trans(a).subs( { a: Matrix([ [1, 2, 3] ]) } )
```